### ISSUE TITLE
error while running 'db init'

### ISSUE THREAD

**Issue by amjed-ali-k** [GalvinGao/gofiber-template#24]
Here is the terminal logs which i get when i run `go run main.go db init`

```
amj@ed:~/Git/server$ go run main.go db init
2023-02-03T09:56:30.868895751+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] SUPPLY      *appconfig.Config evt.name=fx.init
2023-02-03T09:56:30.869027615+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     *bun.DB <= github.com/GalvinGao/gofiber-template/pkg/infra/db.New() from module "db" evt.name=fx.init
2023-02-03T09:56:30.869063165+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     *repo.Post <= github.com/GalvinGao/gofiber-template/pkg/repo.NewPost() from module "repo" evt.name=fx.init
2023-02-03T09:56:30.869095083+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     *service.Post <= github.com/GalvinGao/gofiber-template/pkg/service.NewPost() from module "service" evt.name=fx.init
2023-02-03T09:56:30.869128957+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     *fiber.App <= github.com/GalvinGao/gofiber-template/pkg/server/http.Create() from module "http" evt.name=fx.init
2023-02-03T09:56:30.86915899+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE      fiber.Router[name = "api-v1"] <= github.com/GalvinGao/gofiber-template/pkg/server/http/route.CreateGroups() from module "route" evt.name=fx.init
2023-02-03T09:56:30.869192374+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     fiber.Router[name = "internal"] <= github.com/GalvinGao/gofiber-template/pkg/server/http/route.CreateGroups() from module "route" evt.name=fx.init
2023-02-03T09:56:30.869221499+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     fx.Lifecycle <= go.uber.org/fx.New.func1() evt.name=fx.init
2023-02-03T09:56:30.869247341+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     fx.Shutdowner <= go.uber.org/fx.(*App).shutdowner-fm() evt.name=fx.init
2023-02-03T09:56:30.869276675+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE     fx.DotGraph <= go.uber.org/fx.(*App).dotGraph-fm() evt.name=fx.init
2023-02-03T09:56:30.869312225+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] LOGGER      Initialized custom logger from github.com/GalvinGao/gofiber-template/pkg/x/logger/fxlogger.Logger() evt.name=fx.init
2023-02-03T09:56:30.86934561+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] INVOKE               github.com/GalvinGao/gofiber-template/pkg/controller.RegisterPost() from module "controller" evt.name=fx.init
2023-02-03T09:56:30.869584613+05:30 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] RUNNING evt.name=fx.init
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6e1493]

goroutine 1 [running]:
github.com/uptrace/bun.NewCreateTableQuery(...)
        /home/amj/go/pkg/mod/github.com/uptrace/bun@v1.1.9/query_table_create.go:33
github.com/uptrace/bun.(*DB).NewCreateTable(...)
        /home/amj/go/pkg/mod/github.com/uptrace/bun@v1.1.9/db.go:106
github.com/uptrace/bun/migrate.(*Migrator).Init(0xc0002453b0, {0xb88150, 0xc000124000})
        /home/amj/go/pkg/mod/github.com/uptrace/bun@v1.1.9/migrate/migrator.go:96 +0x73
github.com/GalvinGao/gofiber-template/pkg/infra/db/migrations.(*Migrator).Init(...)
        /home/amj/Git/server/pkg/infra/db/migrations/migrator.go:19
github.com/GalvinGao/gofiber-template/cmd/app/cli/db.Command.func2(0xc00032a040)
        /home/amj/Git/server/cmd/app/cli/db/command.go:38 +0x36
github.com/urfave/cli/v2.(*Command).Run(0xc000274b40, 0xc00032a040, {0xc0003109c0, 0x1, 0x1})
        /home/amj/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/command.go:271 +0xa42
github.com/urfave/cli/v2.(*Command).Run(0xc000274a00, 0xc000147f00, {0xc00027e8e0, 0x2, 0x2})
        /home/amj/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/command.go:264 +0xc97
github.com/urfave/cli/v2.(*Command).Run(0xc0002757c0, 0xc000147e40, {0xc00011c060, 0x3, 0x3})
        /home/amj/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/command.go:264 +0xc97
github.com/urfave/cli/v2.(*App).RunContext(0xc000328000, {0xb88150?, 0xc000124000}, {0xc00011c060, 0x3, 0x3})
        /home/amj/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/app.go:333 +0x665
github.com/urfave/cli/v2.(*App).Run(...)
        /home/amj/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/app.go:310
github.com/GalvinGao/gofiber-template/cmd/app.Run()
        /home/amj/Git/server/cmd/app/app.go:21 +0x106
main.main()
        /home/amj/Git/server/main.go:6 +0x17
exit status 2
```

**Comment by GalvinGao** [comment]
Thanks for reporting the issue. I'll take a look tomorrow.

**Comment by amjed-ali-k** [comment]
B.T.W i am using Ubuntu 22.04.

**Comment by stctheproducer** [comment]
I'm facing the same issue on macOS Monterey.

**Comment by amjed-ali-k** [comment]
@GalvinGao any updates

**Comment by stctheproducer** [comment]
> @GalvinGao any updates

Without updating `github.com/uptrace/bun/dialect/pgdialect` I'm getting

```log
panic: pgdialect and Bun must have the same version: v1.1.10 != v1.1.11

goroutine 1 [running]:
github.com/uptrace/bun/dialect/pgdialect.init.0()
        /Users/stctheproducer/go/pkg/mod/github.com/uptrace/bun/dialect/pgdialect@v1.1.10/dialect.go:20 +0xba
exit status 2
```

After upgrading, I'm getting:

```log
2023-02-11T15:14:52.976631+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] SUPPLY *appconfig.Config evt.name=fx.init
2023-02-11T15:14:52.977454+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        *bun.DB <= github.com/GalvinGao/gofiber-template/pkg/infra/db.New() from module "db" evt.name=fx.init
2023-02-11T15:14:52.97753+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE *repo.Post <= github.com/GalvinGao/gofiber-template/pkg/repo.NewPost() from module "repo" evt.name=fx.init
2023-02-11T15:14:52.977594+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        *service.Post <= github.com/GalvinGao/gofiber-template/pkg/service.NewPost() from module "service" evt.name=fx.init
2023-02-11T15:14:52.977655+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        *fiber.App <= github.com/GalvinGao/gofiber-template/pkg/server/http.Create() from module "http" evt.name=fx.init
2023-02-11T15:14:52.977713+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        fiber.Router[name = "api-v1"] <= github.com/GalvinGao/gofiber-template/pkg/server/http/route.CreateGroups() from module "route" evt.name=fx.init
2023-02-11T15:14:52.977762+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        fiber.Router[name = "internal"] <= github.com/GalvinGao/gofiber-template/pkg/server/http/route.CreateGroups() from module "route" evt.name=fx.init
2023-02-11T15:14:52.978562+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        fx.Lifecycle <= go.uber.org/fx.New.func1() evt.name=fx.init
2023-02-11T15:14:52.978708+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        fx.Shutdowner <= go.uber.org/fx.(*App).shutdowner-fm() evt.name=fx.init
2023-02-11T15:14:52.978814+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] PROVIDE        fx.DotGraph <= go.uber.org/fx.(*App).dotGraph-fm() evt.name=fx.init
2023-02-11T15:14:52.97889+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] LOGGER  Initialized custom logger from github.com/GalvinGao/gofiber-template/pkg/x/logger/fxlogger.Logger() evt.name=fx.init
2023-02-11T15:14:52.978973+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] INVOKE         github.com/GalvinGao/gofiber-template/pkg/controller.RegisterPost() from module "controller" evt.name=fx.init
2023-02-11T15:14:52.992342+02:00 INF pkg/x/logger/fxlogger/fxlogger.go:36 > [Fx] RUNNING evt.name=fx.init
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x12a8b9d]

goroutine 1 [running]:
github.com/uptrace/bun.NewCreateTableQuery(0x0)
        /Users/stctheproducer/go/pkg/mod/github.com/uptrace/bun@v1.1.11/query_table_create.go:42 +0x1d
github.com/uptrace/bun.(*DB).NewCreateTable(...)
        /Users/stctheproducer/go/pkg/mod/github.com/uptrace/bun@v1.1.11/db.go:110
github.com/uptrace/bun/migrate.(*Migrator).Init(0xc00007e190, {0x1798170, 0xc0000ac008})
        /Users/stctheproducer/go/pkg/mod/github.com/uptrace/bun@v1.1.11/migrate/migrator.go:96 +0x49
github.com/GalvinGao/gofiber-template/pkg/infra/db/migrations.(*Migrator).Init(...)
        /Users/stctheproducer/Projects/golang/gofiber-template/pkg/infra/db/migrations/migrator.go:19
github.com/GalvinGao/gofiber-template/cmd/app/cli/db.Command.func2(0xc0000cdf00)
        /Users/stctheproducer/Projects/golang/gofiber-template/cmd/app/cli/db/command.go:38 +0x36
github.com/urfave/cli/v2.(*Command).Run(0xc0001d4b00, 0xc0000cdf00, {0xc0003949c0, 0x1, 0x1})
        /Users/stctheproducer/go/pkg/mod/github.com/urfave/cli/v2@v2.24.3/command.go:273 +0xa42
github.com/urfave/cli/v2.(*Command).Run(0xc0001d49a0, 0xc0000cddc0, {0xc0003826e0, 0x2, 0x2})
        /Users/stctheproducer/go/pkg/mod/github.com/urfave/cli/v2@v2.24.3/command.go:266 +0xc97
github.com/urfave/cli/v2.(*Command).Run(0xc0001d58c0, 0xc0000cdd00, {0xc0000a4090, 0x3, 0x3})
        /Users/stctheproducer/go/pkg/mod/github.com/urfave/cli/v2@v2.24.3/command.go:266 +0xc97
github.com/urfave/cli/v2.(*App).RunContext(0xc0003ec000, {0x1798170?, 0xc0000ac008}, {0xc0000a4090, 0x3, 0x3})
        /Users/stctheproducer/go/pkg/mod/github.com/urfave/cli/v2@v2.24.3/app.go:332 +0x616
github.com/urfave/cli/v2.(*App).Run(...)
        /Users/stctheproducer/go/pkg/mod/github.com/urfave/cli/v2@v2.24.3/app.go:309
github.com/GalvinGao/gofiber-template/cmd/app.Run()
        /Users/stctheproducer/Projects/golang/gofiber-template/cmd/app/app.go:21 +0x106
main.main()
        /Users/stctheproducer/Projects/golang/gofiber-template/main.go:6 +0x17
exit status 2
```

**Comment by stctheproducer** [comment]
@amjed-ali-k @GalvinGao after some debugging I found the issue. I've created a pull request for it: #28

**Comment by GalvinGao** [comment]
#28 is merged! Sorry for the delay and thanks @stctheproducer for the PR! Please see if it resolves the issue :D

**Comment by stctheproducer** [comment]
You're welcome! I just tested a fresh clone and the only issue remaining is upgrading `pgdialect` to `v1.1.11`

**Comment by GalvinGao** [comment]
> You're welcome! I just tested a fresh clone and the only issue remaining is upgrading `pgdialect` to `v1.1.11`

Should be fixed in 0850bbafd2f98f3e79210d5c93a0b86ccea88239 :D

**Comment by GalvinGao** [comment]
Hi @amjed-ali-k just want to check in to see if this actually got fixed for you :D; if so I'm gonna close this issue.
